File: app/src/main/java/org/my/alias/UI/ScoreDialog.java
package org.my.alias.UI;

import android.annotation.SuppressLint;
import android.app.Dialog;
import androidx.fragment.app.DialogFragment;
import android.content.DialogInterface;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import androidx.appcompat.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

import org.my.alias.Preferences;
import org.my.alias.R;
import org.my.alias.Team;
import org.my.alias.UI.GameScreens.StartScreen;

import java.util.ArrayList;

import br.com.kots.mob.complex.preferences.ComplexPreferences;
import butterknife.BindView;
import butterknife.ButterKnife;

public class ScoreDialog extends DialogFragment {
    String SAVE_DATA_KEY = "score";
    ArrayList<Team> allTeams;
    @BindView (R.id.first_team)TextView textFirst;
    @BindView (R.id.second_team)TextView textSecond;
    @BindView (R.id.thurd_team)TextView textThird;
    @BindView (R.id.fourth_team)TextView textFourth;


    public ScoreDialog(){
        super();
    }

    @SuppressLint("ValidFragment")
    public ScoreDialog(ArrayList<Team> allTeams){
        super();
        this.allTeams = allTeams;
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putSerializable(SAVE_DATA_KEY, allTeams);
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getActivity());
        ComplexPreferences complexPreferences = ComplexPreferences.getComplexPreferences(getActivity(), "all_teams", getActivity().MODE_PRIVATE);
        LayoutInflater inflater = LayoutInflater.from(getActivity());
        View v = inflater.inflate(R.layout.dialog_layout, null);
        ButterKnife.bind(this, v);
        Team firstTeam;
        Team secondTeam;
        Team thirdTeam;
        Team fourthTeam;
        int teamsNumber = Integer.parseInt(sharedPreferences.getString(Preferences.KEY_QANTITY_TEAMS_FROM_PREFS, "2"));
        switch (teamsNumber) {
            case 2:
                firstTeam = complexPreferences.getObject(Preferences.KEY_FIRST, Team.class);
                textFirst.setText(getString(R.string.perform_score, getString(R.string.team1),
                        String.valueOf(firstTeam.getScore()), String.valueOf(firstTeam.getSteps())));
                secondTeam = complexPreferences.getObject(Preferences.KEY_SECOND, Team.class);
                textSecond.setText(getString(R.string.perform_score, getString(R.string.team2),
                        String.valueOf(secondTeam.getScore()), String.valueOf(secondTeam.getSteps())));
                textThird.setVisibility(View.GONE);
                textFourth.setVisibility(View.GONE);
                break;
            case 3:
                firstTeam = complexPreferences.getObject(Preferences.KEY_FIRST, Team.class);
                textFirst.setText(getString(R.string.perform_score, getString(R.string.team1),
                        String.valueOf(firstTeam.getScore()), String.valueOf(firstTeam.getSteps())));
                secondTeam = complexPreferences.getObject(Preferences.KEY_SECOND, Team.class);
                textSecond.setText(getString(R.string.perform_score, getString(R.string.team2),
                        String.valueOf(secondTeam.getScore()), String.valueOf(secondTeam.getSteps())));
                thirdTeam = complexPreferences.getObject(Preferences.KEY_THIRD, Team.class);
                textThird.setText(getString(R.string.perform_score, getString(R.string.team3),
                        String.valueOf(thirdTeam.getScore()), String.valueOf(thirdTeam.getSteps())));
                textFourth.setVisibility(View.GONE);
                break;
            case 4:
                firstTeam = complexPreferences.getObject(Preferences.KEY_FIRST, Team.class);
                textFirst.setText(getString(R.string.perform_score, getString(R.string.team1),
                        String.valueOf(firstTeam.getScore()), String.valueOf(firstTeam.getSteps())));
                secondTeam = complexPreferences.getObject(Preferences.KEY_SECOND, Team.class);
                textSecond.setText(getString(R.string.perform_score, getString(R.string.team2),
                        String.valueOf(secondTeam.getScore()), String.valueOf(secondTeam.getSteps())));
                thirdTeam = complexPreferences.getObject(Preferences.KEY_THIRD, Team.class);
                textThird.setText(getString(R.string.perform_score, getString(R.string.team3),
                        String.valueOf(thirdTeam.getScore()), String.valueOf(thirdTeam.getSteps())));
                fourthTeam = complexPreferences.getObject(Preferences.KEY_FOURTH, Team.class);
                textFourth.setText(getString(R.string.perform_score, getString(R.string.team4),
                        String.valueOf(fourthTeam.getScore()), String.valueOf(fourthTeam.getSteps())));
                break;
        }

        AlertDialog score = new AlertDialog.Builder(getActivity()).setView(v).setTitle(getString(R.string.score))
                .setInverseBackgroundForced(true)
                .setPositiveButton(getString(R.string.ok), new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        dismiss();
                    }
                }).setNegativeButton(getString(R.string.next_round), new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        Intent intent = new Intent(getActivity(), StartScreen.class);
                        intent.getBooleanExtra("continue", true);
                        startActivity(intent);
                    }
                }).create();
        return score;
    }
}

File: app/src/main/java/org/my/alias/UI/ScoreDialog.kt
package org.my.alias.UI

import android.annotation.SuppressLint
import android.app.Dialog
import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.preference.PreferenceManager
import android.view.LayoutInflater
import android.view.View
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import br.com.kots.mob.complex.preferences.ComplexPreferences
import kotlinx.android.synthetic.main.dialog_layout.view.*
import org.my.alias.Preferences
import org.my.alias.Preferences.Companion.KEY_CONTINUE
import org.my.alias.R
import org.my.alias.Team
import org.my.alias.UI.GameScreens.StartScreen
import java.util.*


class ScoreDialog : DialogFragment {
    internal var SAVE_DATA_KEY = "score"
    internal lateinit var allTeams: ArrayList<Team>
    lateinit var sharedPreferences : SharedPreferences
    lateinit var complexPreferences : ComplexPreferences
    lateinit var firstTeam: Team
    lateinit var secondTeam: Team
    lateinit var thirdTeam: Team
    lateinit var fourthTeam: Team

    constructor() : super() {}

    @SuppressLint("ValidFragment")
    constructor(allTeams: ArrayList<Team>) : super() {
        this.allTeams = allTeams
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putSerializable(SAVE_DATA_KEY, allTeams)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(activity)
        complexPreferences = ComplexPreferences.getComplexPreferences(activity, "all_teams", Context.MODE_PRIVATE)
        val inflater = LayoutInflater.from(activity)
        val v = inflater.inflate(R.layout.dialog_layout, null)
        confifUi(v);
        return AlertDialog.Builder(activity!!).setView(v).setTitle(getString(R.string.score))
                .setInverseBackgroundForced(true)
                .setPositiveButton(getString(R.string.ok)) { dialog, which -> dismiss() }.setNegativeButton(getString(R.string.next_round)) { dialog, which ->
                    val intent = Intent(activity, StartScreen::class.java)
                    intent.getBooleanExtra(KEY_CONTINUE, true)
                    startActivity(intent)
                }.create()
    }

    private fun confifUi(view: View) {
        val teamsNumber = Integer.parseInt(sharedPreferences.getString(Preferences.KEY_QANTITY_TEAMS_FROM_PREFS, "2")!!)
        when (teamsNumber) {
            2 -> {
                firstTeam = complexPreferences.getObject<Team>(Preferences.KEY_FIRST, Team::class.java)
                view.first_team.text = getString(R.string.perform_score, getString(R.string.team1),
                        firstTeam.score.toString(), firstTeam.steps.toString())
                secondTeam = complexPreferences.getObject<Team>(Preferences.KEY_SECOND, Team::class.java)
                view.second_team.text = getString(R.string.perform_score, getString(R.string.team2),
                        secondTeam.score.toString(), secondTeam.steps.toString())
                view.thurd_team.visibility = View.GONE
                view.fourth_team.visibility = View.GONE
            }
            3 -> {
                firstTeam = complexPreferences.getObject<Team>(Preferences.KEY_FIRST, Team::class.java)
                view.first_team.text = getString(R.string.perform_score, getString(R.string.team1),
                        firstTeam.score.toString(), firstTeam.steps.toString())
                secondTeam = complexPreferences.getObject<Team>(Preferences.KEY_SECOND, Team::class.java)
                view.second_team.text = getString(R.string.perform_score, getString(R.string.team2),
                        secondTeam.score.toString(), secondTeam.steps.toString())
                thirdTeam = complexPreferences.getObject<Team>(Preferences.KEY_THIRD, Team::class.java)
                view.thurd_team.text = getString(R.string.perform_score, getString(R.string.team3),
                        thirdTeam.score.toString(), thirdTeam.steps.toString())
                view.fourth_team.visibility = View.GONE
            }
            4 -> {
                firstTeam = complexPreferences.getObject<Team>(Preferences.KEY_FIRST, Team::class.java)
                view.first_team.text = getString(R.string.perform_score, getString(R.string.team1),
                        firstTeam.score.toString(), firstTeam.steps.toString())
                secondTeam = complexPreferences.getObject<Team>(Preferences.KEY_SECOND, Team::class.java)
                view.second_team.text = getString(R.string.perform_score, getString(R.string.team2),
                        secondTeam.score.toString(), secondTeam.steps.toString())
                thirdTeam = complexPreferences.getObject<Team>(Preferences.KEY_THIRD, Team::class.java)
                view.thurd_team.text = getString(R.string.perform_score, getString(R.string.team3),
                        thirdTeam.score.toString(), thirdTeam.steps.toString())
                fourthTeam = complexPreferences.getObject<Team>(Preferences.KEY_FOURTH, Team::class.java)
                view.fourth_team.text = getString(R.string.perform_score, getString(R.string.team4),
                        fourthTeam.score.toString(), fourthTeam.steps.toString())
            }
        }
    }

}
